pyqir-generator/README.md

```diff
@@ -1,22 +1,102 @@
-# pyqir-generator
+# pyqir_generator
 
-## Building and Testing
+The `pyqir_generator` package provides the ability to generate [QIR](https://github.com/microsoft/qsharp-language/tree/main/Specifications/QIR#quantum-intermediate-representation-qir) using a [builder](https://en.wikipedia.org/wiki/Builder_pattern) API.
+
+It is intended to be used by code automating translation processes enabling the conversion in some format to QIR via Python.
+
+## Examples
+
+There are generator examples in the repository:
+- [Bernstein-Vazirani](examples/generator/bernstein_vazirani.py)
+- [Superposition](examples/generator/superposition.py)
+
+A short example is below.
+
+This operation sets two qubits in superposition and returns the result of 
+measuring them. The full output is sent to the terminal:
+
+```python
+from pyqir_generator import QirBuilder
 
-To build this package, first install `maturin`:
+"""
+Whether the two qubits should be entangled.
+    If "true", the two qubits are entangled such that the state of both
+    is one of the Bell states: | x 〉| y 〉 = 1/sqrt(2) [|0〉|0〉 + |1〉|1〉]
+    If "false", the two qubits are entangled such that the state is:
+    | x 〉| y 〉 = 1/4 [|0〉|0〉 + |0〉|1〉 + |1〉|0〉 + |1〉|1〉]
+"""
+entangle: bool = True
 
-```shell
-pip install maturin
+builder = QirBuilder("Bell")
+builder.add_quantum_register("qubit", 2)
```

Comment:
My understanding may be wrong, but I believe this function allocates/generates/reserves 2 quantum registers with names `qubit0` and `qubit1`? If so, I would suggest that the rename is made plural `add_quantum_registers`. One could optionally make an `add_quantum_register("qubit")` which would register `qubit` with no numbers appended.